### PR TITLE
Update dependencies, refactor deprecated methods

### PR DESCRIPTION
**Pull request by RomanPolach**
Update dependencies, refactor deprecated methods: 

updated dependencies:

AtomicFU -  "0.19.0"
Koin - "3.3.2"
AndroidX  Core - "1.9.0"
Android Material - 1.8.0-rc01
Android lifecycle -  "2.5.1"
Compose - 1.3.3
Android Activity - "1.6.1"
Android Navigation Compose -  "2.5.3"
Constraint Layout Compose - "1.0.1"
Accompanist =  "0.28.0"
 Ktor -  "2.2.2"
Kermit -  "1.2.2"
SQL Delight - "1.5.5"
Google Play - "21.0.1"
Google Location - 18.0.0"
Kotlin (stdlib)  - "1.8.0"

### CODE REVIEW

**Comment by pchmelar** [comment]
KMP build is now broken 😢 
https://github.com/MateeDevs/devstack-native-app/actions/runs/3948180679/jobs/6757875341

File: build.gradle.kts [hunk 1]
```diff
@@ -10,7 +10,7 @@ buildscript {
 
     dependencies {
         classpath("org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlinVersion")
-        classpath("com.android.tools.build:gradle:7.2.0")
+        classpath("com.android.tools.build:gradle:7.0.4")
```
Comment (vojta-horanek) [inline build.gradle.kts]: Downgrade?

File: buildSrc/build.gradle.kts [hunk 2]
```diff
@@ -1,4 +1,4 @@
-val kotlinVersion = "1.7.10"
+val kotlinVersion = "1.7.20"
 val androidGradleVersion = "7.2.0"
```
Comment (vojta-horanek) [inline buildSrc/build.gradle.kts:2]: + the "old" version is still here

File: buildSrc/build.gradle.kts [hunk 3]
```diff
@@ -1,4 +1,4 @@
-val kotlinVersion = "1.7.10"
+val kotlinVersion = "1.7.20"
```
Comment (vojta-horanek) [inline buildSrc/build.gradle.kts]: Kotlin 1.7.20

File: buildSrc/src/main/kotlin/Dependency.kt [hunk 4]
```diff
@@ -3,10 +3,11 @@
 object Dependency {
 
     object Kotlin {
-        const val stdlib = "org.jetbrains.kotlin:kotlin-stdlib:$kotlinVersion"
+        const val version = "1.8.0"
```
Comment (vojta-horanek) [inline buildSrc/src/main/kotlin/Dependency.kt]: Kotlin 1.8

File: buildSrc/src/main/kotlin/Dependency.kt [hunk 5]
```diff
@@ -18,78 +19,78 @@ object Dependency {
...
             const val runtime = "androidx.paging:paging-runtime:$version"
             const val compose = "androidx.paging:paging-compose:$composeVersion"
         }
     }
 
     object Compose {
-        const val version = "1.3.0-rc01"
-        const val compilerExtensionVersion = version
+        const val version = "1.3.3"
+        const val compilerExtensionVersion = "1.3.2"
```
Comment (vojta-horanek) [inline buildSrc/src/main/kotlin/Dependency.kt]: 1.3.2 is [not compatible](https://developer.android.com/jetpack/androidx/releases/compose-kotlin) with Kotlin 1.8. I'm not sure which kotlin version you wanted to upgrade to tho...

File: buildSrc/src/main/kotlin/Dependency.kt [hunk 6]
```diff
@@ -18,78 +19,78 @@ object Dependency {
...
 
         const val ui = "androidx.compose.ui:ui:$version"
         const val uiTooling = "androidx.compose.ui:ui-tooling:$version"
-        const val foundation = "androidx.compose.foundation:foundation:$version"
+        const val foundation = "androidx.compose.foundation:foundation:1.3.1"
```
Comment (vojta-horanek) [inline buildSrc/src/main/kotlin/Dependency.kt:65]: Why are you hardcoding the versions here? The version constant should really be used instead

File: buildSrc/src/main/kotlin/Dependency.kt [hunk 7]
```diff
@@ -146,8 +147,8 @@ object Dependency {
     }
 
     object Play {
-        private const val version = "18.0.0"
+        private const val version = "21.0.1"
 
-        const val location = "com.google.android.gms:play-services-location:$version"
+        const val location = "com.google.android.gms:play-services-location:18.0.0"
```
Comment (vojta-horanek) [inline buildSrc/src/main/kotlin/Dependency.kt]: You've updated the constant but then made it not used? This does not update gms location

File: buildSrc/src/main/kotlin/Dependency.kt [hunk 8]
```diff
@@ -3,10 +3,11 @@
 object Dependency {
 
     object Kotlin {
-        const val stdlib = "org.jetbrains.kotlin:kotlin-stdlib:$kotlinVersion"
+        const val version = "1.8.0"
+        const val stdlib = "org.jetbrains.kotlin:kotlin-bom:$version"
```
Comment (vojta-horanek) [inline buildSrc/src/main/kotlin/Dependency.kt]: This is not a stdlib now is it? Bom just makes sure all the version of kotlin libs match.

**Review by vojta-horanek — CHANGES_REQUESTED** [review]
Mixed up kotlin versions & AGP versions

File: buildSrc/src/main/kotlin/Dependency.kt (hunk 4 above)
Comment (RomanPolach) [reply buildSrc/src/main/kotlin/Dependency.kt]: Já vím. Já jsem chtěl mít všude tu 1.7.20, ale když to nastavím pro tu "org.jetbrains.kotlin:kotlin-bom:$version" knihovnu, tak to nejde zbuildit...Mmt, napíšu ti zprávu

File: buildSrc/src/main/kotlin/Dependency.kt (hunk 6 above)
Comment (RomanPolach) [reply buildSrc/src/main/kotlin/Dependency.kt:65]: Protože ta konstanta je 1.3.3, ale některé součásti compose jsou jen 1.3.1. Tak mi to přišlo lepší to tu napsat explicitně

Comment (vojta-horanek) [reply buildSrc/src/main/kotlin/Dependency.kt:65]: Aha, už to vidím. Hlavně aby se to tu pak nenechalo hardcodenuté.

**Review by vojta-horanek — APPROVED** [review]

**Pull request merged**